xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/main/java/org/xwiki/rendering/internal/macro/include/IncludeMacro.java

```diff
@@ -212,7 +213,20 @@ public void setDocumentDisplayer(DocumentDisplayer documentDisplayer)
                 references.pop();
             }
         }
-
+        
+        // Exclude the first heading from the content
+        if (parameters.excludeFirstHeading()) {
+            HeaderBlock heading =
+                    result.getFirstBlock(new ClassBlockMatcher(HeaderBlock.class), Block.Axes.DESCENDANT_OR_SELF);
+            if (heading != null) {
+                result.removeBlock(heading);
+            } else {
+                String includeType = parameters.getSection() != null ? "section" : "document";
+                throw new MacroExecutionException(
+                    String.format("The included [%s] doesn't contain any heading",includeType));
```

Comment:
Note: bad code style, missing a space after the command. You should apply code style across the full file and make it part of your IDE code style to make sure to not forget it.